src/drive/web/modules/services/components/Embeder.jsx

```diff
@@ -40,7 +43,20 @@ class Embeder extends React.Component {
           <pre className="u-error">{this.state.error.toString()}</pre>
         )}
         {this.state.fileId && (
-          <FileOpenerExternal service={service} fileId={this.state.fileId} />
+          <Router history={hashHistory}>
+            <Route
+              path="/"
+              component={() => (
+                <FileOpenerExternal
+                  service={service}
+                  fileId={this.state.fileId}
+                />
+              )}
+            />
+            {isOnlyOfficeEnabled() && (
+              <Route path="onlyoffice/:fileId" component={OnlyOfficeView} />
+            )}
+          </Router>
```

Comment:
I don't understand. How can we arrive on this component but also with onlyoffice/:fileId in the URL?

Reply:
I got it. We display the viewer with the "open with OO" button. And then when the user click on it, we go to the right route. Is it the wanted behavior? Don't we want to display the OO Editor directly?

Reply:
Oui c'est ça effectivement ! Oui c'est le comportement attendu (donc pas d'ouverture direct du fichier) pour l'instant. Il y a une carte spécifiquement pour la gestion du partage. Ici ça ne concerne que la visionneuse.

Reply:
A terme ça devrait s'ouvrir directement. Ce n'est pas nécessaire dans cette PR (vu avec @poupotte )